sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/KeyClientImpl.java

```diff
@@ -0,0 +1,1420 @@
+        Objects.requireNonNull(keyProperties, "The key properties input parameter cannot be null.");
+        KeyRequestParameters parameters = new KeyRequestParameters()
+            .setTags(keyProperties.getTags())
+            .setKeyAttributes(new KeyRequestAttributes(keyProperties))
+            .setReleasePolicy(keyProperties.getReleasePolicy());
+
+        if (keyOperations.length > 0) {
+            parameters.setKeyOps(Arrays.asList(keyOperations));
+        }
+        return parameters;
+    }
+
+    public PollerFlux<DeletedKey, Void> beginDeleteKeyAsync(String name) {
+        return new PollerFlux<>(getDefaultPollingInterval(),
+            activationOperation(name),
+            createPollOperation(name),
+            (context, firstResponse) -> Mono.empty(),
+            (context) -> Mono.empty());
+    }
+
+    private Function<PollingContext<DeletedKey>, Mono<DeletedKey>> activationOperation(String name) {
+        return (pollingContext) -> withContext(context -> deleteKeyWithResponse(name, context))
+            .flatMap(deletedKeyResponse -> Mono.just(deletedKeyResponse.getValue()));
+    }
+
+    /*
+     * Polling operation to poll on create delete key operation status.
+     */
+    private Function<PollingContext<DeletedKey>, Mono<PollResponse<DeletedKey>>> createPollOperation(String keyName) {
+        return pollingContext ->
+            withContext(context -> service.getDeletedKeyPollerAsync(vaultUrl, keyName, keyServiceVersion.getVersion(),
+                ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE)))
+                .flatMap(deletedKeyResponse -> {
+                    if (deletedKeyResponse.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+                        return Mono.defer(() -> Mono.just(new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS,
+                            pollingContext.getLatestResponse().getValue())));
+                    }
+                    return Mono.defer(() ->
+                        Mono.just(new PollResponse<>(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED,
+                            deletedKeyResponse.getValue())));
+                })
+                // This means either vault has soft-delete disabled or permission is not granted for the get deleted key
+                // operation. In both cases deletion operation was successful when activation operation succeeded before
+                // reaching here.
+                .onErrorReturn(new PollResponse<>(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED,
+                    pollingContext.getLatestResponse().getValue()));
+    }
+
+    private Mono<Response<DeletedKey>> deleteKeyWithResponse(String name, Context context) {
+        context = enableSyncRestProxy(context);
+        return service.deleteKeyAsync(vaultUrl, name, keyServiceVersion.getVersion(), ACCEPT_LANGUAGE,
```

Comment:
Should this be calling deleteKey instead?

Reply:
nope, it is used by poller, so has to be async for now.